### PR TITLE
test: integration test eoa tx mined

### PR DESCRIPTION
**Pull request by BatiGencho**
This MR covers integration tests for our poa.

### CODE REVIEW

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            temp_path: tempfile::TempDir::new().expect("tempdir is okay").into_path(),
+            secret_key,
+            rpc_port: RPC_PORT_BASE + index,
+            authrpc_port: AUTHRPC_PORT_BASE + index,
+            discovery_port: DISCOVERY_PORT_BASE + index,
+            peers_list: vec![],
+        }
+    }
+
+    pub fn insert_piers_list(&mut self, peers: Vec<FederationMemberTestConfig>) {
```
Comment (arminsabouri) [inline crates/consensus/authority/tests/it/poa_testnet.rs]: Just a little typo here

```suggestion
    pub fn insert_peers_list(&mut self, peers: Vec<FederationMemberTestConfig>) {
```

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+            index,
+            temp_path: tempfile::TempDir::new().expect("tempdir is okay").into_path(),
+            secret_key,
+            rpc_port: RPC_PORT_BASE + index,
+            authrpc_port: AUTHRPC_PORT_BASE + index,
+            discovery_port: DISCOVERY_PORT_BASE + index,
+            peers_list: vec![],
+        }
+    }
+
```
Comment (scottmillner) [inline crates/consensus/authority/tests/it/poa_testnet.rs:162]: nit `insert_piers_list` > `insert_peers_list`

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    pub fn insert_piers_list(&mut self, peers: Vec<FederationMemberTestConfig>) {
+        self.peers_list = peers;
+    }
+
+    pub fn build_command(&self) -> PoaNodeCommand<NoArgsCliExt<FederationMemberTestConfig>> {
+        println!("Engine {} data directory", self.index);
+        println!(
```
Comment (scottmillner) [inline crates/consensus/authority/tests/it/poa_testnet.rs:169]: nit `secrete` > `secret`

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    }
+}
+
+impl RethNodeCommandConfig for FederationMemberTestConfig {
+    fn on_node_started<Reth: RethNodeComponents>(&mut self, components: &Reth) -> eyre::Result<()> {
+        let engine_index = self.index;
+        println!("Engine {} started task", engine_index);
+
+        // add the peers
+        for peer in self.peers_list.iter() {
```
Comment (scottmillner) [inline crates/consensus/authority/tests/it/poa_testnet.rs:228]: nit `pier` > `peer`

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+
+fn create_poa_federation_members() -> HashMap<usize, FederationMemberTestConfig> {
+    // create two secret keys one for each member
+    let sc1 = FED_MEMBER1_SECRET_KEY.parse::<SecretKey>().unwrap();
+    let sc2 = FED_MEMBER2_SECRET_KEY.parse::<SecretKey>().unwrap();
+
+    // create the member configs
+    let mut fed_member_config1 = FederationMemberTestConfig::new(0, sc1);
+    let mut fed_member_config2 = FederationMemberTestConfig::new(1, sc2);
+
```
Comment (scottmillner) [inline crates/consensus/authority/tests/it/poa_testnet.rs:354]: nit `piers` > `peers`

**Review by scottmillner — APPROVED** [review]
awesome sauce!! 🔥 🔥 🔥 

just added some comments around typos

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 2 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs:162]: well spotted thanks :)

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 3 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs:169]: well spotted thanks :) fixed

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 4 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs:228]: well spotted thanks :)  fixed

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 5 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs:354]: well spotted thanks :) fixed

File: crates/consensus/authority/tests/it/poa_testnet.rs [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+        // send eoa message to first node
+        println!("Sending eoa message...");
+        let _ = send_botanix(
+            &targeted_fed_member,
+            SENDER_SECRET_KEY,
+            RECEIVER_ADDRESS,
+            U256::from(1000),
+        )
+        .await; // 1 ETH = BTC in wei
+        tokio::time::sleep(Duration::from_secs(30)).await;
```
Comment (arminsabouri) [inline crates/consensus/authority/tests/it/poa_testnet.rs]: When do we break out of this loop?

**Review by arminsabouri — APPROVED** [review]

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 1 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs]: fixed

File: crates/consensus/authority/tests/it/poa_testnet.rs (hunk 6 above)
Comment (BatiGencho) [reply crates/consensus/authority/tests/it/poa_testnet.rs]: I think this commetn is outdated. We stop after 3 rounds now.

**Pull request merged**